### PR TITLE
Demonstrate how to make grpc_library use google.golang.org/protobuf

### PR DESCRIPTION
**Pull request by linuxerwang**
instead of github.com/golang/protubuf.

Not aimed for merge, because it's not backward compatible.

See issue #1485 for the context.

### CODE REVIEW

**Comment by Tatskaari** [comment]
Amazing! Thanks for this! I think to make this backwards compatible, could you introduce a feature flag and only add the new plugin tools and flags if it has been set? See the bottom of `type Configuration struct {}` in `config.go`. 

If you don't have time, I can take over.

**Comment by linuxerwang** [comment]
@Tatskaari Please take it over. I only have time at weekend, and I am not really familiar with Please's internals. This PR is just a demo.  Thank you!

File: src/core/config.go [hunk 1]
```diff
@@ -525,6 +526,7 @@ type Configuration struct {
 		ProtocGoPlugin   string   `help:"The binary passed to protoc as a plugin to generate Go code. Defaults to protoc-gen-go.\nWe've found this easier to manage with a go_get rule instead though, so you can also pass a build label here. See the Please repo for an example." var:"PROTOC_GO_PLUGIN"`
 		GrpcPythonPlugin string   `help:"The plugin invoked to compile Python code for grpc_library.\nDefaults to protoc-gen-grpc-python." var:"GRPC_PYTHON_PLUGIN"`
 		GrpcJavaPlugin   string   `help:"The plugin invoked to compile Java code for grpc_library.\nDefaults to protoc-gen-grpc-java." var:"GRPC_JAVA_PLUGIN"`
+		GrpcGoPlugin     string   `help:"The plugin invoked to compile Go code for grpc_library.\nDefaults to protoc-gen-go-grpc." var:"GRPC_GO_PLUGIN"`
```
Comment (peterebden) [inline src/core/config.go:529]: Should we skip the feature flag and just operate on this config value? Default it to empty, but use it if it's set?

Comment (Tatskaari) [reply src/core/config.go:528]: Yeah I guess I wanted all the config options here to behave the same (i.e. look for the plugin on the path) but go seems to behave differently to the other languages anyway.

**Pull request merged**